lib/membrane_rtc_engine/engine.ex

```diff
@@ -295,11 +299,16 @@ defmodule Membrane.RTC.Engine do
           opts :: endpoint_options_t()
         ) :: :ok | :error
   def add_endpoint(pid, endpoint, opts) do
-    if Keyword.has_key?(opts, :endpoint_id) and Keyword.has_key?(opts, :peer_id) do
-      :error
-    else
-      send(pid, {:add_endpoint, endpoint, opts})
-      :ok
+    cond do
+      Keyword.has_key?(opts, :endpoint_id) and Keyword.has_key?(opts, :peer_id) ->
+        raise "You can't pass both option endpoint_id and peer_id"
+
+      not Keyword.has_key?(opts, :endpoint_id) and not Keyword.has_key?(opts, :peer_id) ->
+        raise "You didn't pass one option endpoint_id or peer_id"
```

Comment:
This sounds a little strange to me.
Also, I don't think that docs for this function mention that you need to supply exactly one, only that they are mutually exclusive.